contracts/HomeBridge.sol

```diff
@@ -1,111 +1,54 @@
-        require(requiredSignatures <= v.length);
-
-        for (uint256 i = 0; i < requiredSignatures; i++) {
-            var a = ecrecover(hash, v[i], r[i], s[i]);
-            require(Helpers.addressArrayContains(authorities, a));
-            require(!Helpers.addressArrayContains(used, a));
-            used[i] = a;
-        }
-        _;
-    }
-
-    /// Constructor.
-    function HomeBridge(
-        uint256 requiredSignaturesParam,
-        address[] authoritiesParam,
-        uint256 estimatedGasCostOfWithdrawParam
-    ) public
-    {
-        require(requiredSignaturesParam != 0);
-        require(requiredSignaturesParam <= authoritiesParam.length);
-        requiredSignatures = requiredSignaturesParam;
-        authorities = authoritiesParam;
-        estimatedGasCostOfWithdraw = estimatedGasCostOfWithdrawParam;
+    function HomeBridge (
+        address _validatorContract
+    ) public Validatable(_validatorContract) {
     }
 
     /// Should be used to deposit money.
     function () public payable {
         Deposit(msg.sender, msg.value);
     }
 
-    /// to be called by authorities to check
-    /// whether they withdraw message should be relayed or whether it
-    /// is too low to cover the cost of calling withdraw and can be ignored
-    function isMessageValueSufficientToCoverRelay(bytes message) public view returns (bool) {
-        return Message.getValue(message) > getWithdrawRelayCost();
+    function setGasLimitWithdrawRelay(uint256 _gas) public onlyValidator {
+        gasLimitWithdrawRelay = _gas;
+        GasConsumptionLimitsUpdated(gasLimitWithdrawRelay);
     }
 
-    /// an upper bound to the cost of relaying a withdraw by calling HomeBridge.withdraw
-    function getWithdrawRelayCost() public view returns (uint256) {
-        return estimatedGasCostOfWithdraw.mul(tx.gasprice);
-    }
+    function withdraw(uint8[] vs, bytes32[] rs, bytes32[] ss, bytes message) public {
+        require(message.length == 116);
+        // require(Helpers.hasEnoughValidSignatures(message, vs, rs, ss, validatorContract));
 
-    /// Used to withdraw money from the contract.
-    ///
-    /// message contains:
-    /// withdrawal recipient (bytes20)
-    /// withdrawal value (uint)
-    /// foreign transaction hash (bytes32) // to avoid transaction duplication
-    ///
-    /// NOTE that anyone can call withdraw provided they have the message and required signatures!
-    function withdraw(uint8[] v, bytes32[] r, bytes32[] s, bytes message) public allAuthorities(v, r, s, message) {
-        require(message.length == 84);
         address recipient = Message.getRecipient(message);
-        uint value = Message.getValue(message);
+        uint256 value = Message.getValue(message);
         bytes32 hash = Message.getTransactionHash(message);
-
-        // The following two statements guard against reentry into this function.
-        // Duplicated withdraw or reentry.
+        uint256 homeGasPrice = Message.getHomeGasPrice(message);
```

Comment:
Let's remove all `homeGasPrice` related logic and work with `value` only

Reply:
why?

Reply:
it does not make sense to cover expenses of bridge validators on Home side since Home is POA network and all transactions are "free" here.

Reply:
you don't get it, it is not the only use case here, people can also withdraw if they provide the corrent gasPrice value